### PR TITLE
Nominatim 4.1.0

### PR DESCRIPTION
**Pull request by philipkozeny**
- updated nominatim to 4.1
- updated base image to 22.04
- therefore also switched to PostgreSQL 14
- removed the tokenizer hack
- optimised graceful shutdowns
- removed 3.7 from currently supported versions
- added `nominatim admin warm` command after initial import

### CODE REVIEW

**Comment by philipkozeny** [comment]
@leonardehrenfried I think I am done with all the required changes and cleanup. The tests are working, so everything should be fine. Maybe you have time for a quick review, thanks!

File: 4.1/README.md [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+  - [Assorted use cases documented in issues](#assorted-use-cases-documented-in-issues)
+
+***
+
+## Automatic import
+
+Download the required data, initialize the database and start nominatim in one go
+
+```
+docker run -it --rm \
```
Comment (leonardehrenfried) [inline 4.1/README.md:27]: I wonder if we should keep the `--rm` in the instructions. We keep getting complaints from people who don't know what it means.

WDYT?

Comment (philipkozeny) [reply 4.1/README.md:27]: Good point, I always tend to forget that too many people just copy & paste our instructions without knowing what they do :)

File: 4.1/contrib/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+services:
+    nominatim:
+        container_name: nominatim
+        image: mediagis/nominatim:4.1
+        restart: always
+        ports:
+            - "8080:8080"
+        environment:
+            # see https://github.com/mediagis/nominatim-docker/tree/master/4.0#configuration for more options
```
Comment (leonardehrenfried) [inline 4.1/contrib/docker-compose.yml:11]:
```suggestion
# see https://github.com/mediagis/nominatim-docker/tree/master/4.1#configuration for more options
```

**Review by leonardehrenfried — APPROVED** [review]
This looks like solid work. Thanks!

**Comment by philipkozeny** [comment]
Triggered it with a small readme change

**Pull request merged**